lib/betagouvbot.rb

```diff
@@ -1,7 +1,7 @@
 # encoding: utf-8
 # frozen_string_literal: true
 
-Dir[__dir__ + '/betagouvbot/**/*.rb'].sort.each(&method(:require))
+Dir[__dir__ + '/betagouvbot/**/*.rb'].each(&method(:require))
```

Comment:
> My gut objects to this very strongly.

Gut feeling changes with mood, language choice, etc.

> We found out that the behaviour of Dir.each is platform-dependent, "thanks" to production downtime. The sort was introduced to restore deterministic behaviour here.

I'm not against the sort. This PR attacks the problem with another angle, though: let's learn to when handling domain objects in constants, be sure they're available to us, or to explicitly pass a block to them if we're not.

> Taking it out feels like playing Russian Roulette with the next similar problem we introduce without our tests catching it owing to unlucky coincidence.

Or better design 😃, which is what I'm proposing.

> Now that I've had a chance to examine this line, I also find myself objecting to it, I'd prefer explicit (and minimal) statements of dependencies, a habit from my Elm and Haskell forays.

Everyone has its tastes.

In Ruby, requires remain during runtime, so requiring the same dependence more than once is wasteful.

Than being said, it is perfectly valid as an alternative solution to do an explicit global loading.